ui/src/app/job-list/table/table.component.ts

```diff
@@ -249,4 +261,59 @@ export class JobsTableComponent implements OnInit {
       }
     }
   }
+
+  openbulkEditDialog(): void {
+    this.resetBulkLabelFieldDefaults();
+    // figure out default values for bulk fields; if all jobs have the same label value,
+    // set that to the default; otherwise, set default to boolean false
+    for (let job of this.selection.selected) {
+      for (let bulkFieldItem of this.bulkLabelFields) {
+        const label = bulkFieldItem.displayField.field.replace('labels.', '');
+        const jobLabelValue = job.labels[label] || '';
+        if (bulkFieldItem.default === null) {
+          bulkFieldItem.default = jobLabelValue;
+        } else if (bulkFieldItem.default !== false && bulkFieldItem.default !== jobLabelValue) {
+          bulkFieldItem.default = false;
+        }
+      }
+    }
+    let dialogRef = this.bulkEditDialog.open(JobsBulkEditComponent, {
+      disableClose: true,
+      data: {
+        'bulkLabelFields' : this.bulkLabelFields,
+        'selectedJobs' :  this.selection.selected,
+        'newValues': []
+      }
+    });
+
+    dialogRef.afterClosed().subscribe((result) => {
+      if (result) {
```

Comment:
What happens if the user cancels or doesn't change any values?

Reply:
if the user cancels, then `result` will be empty because `saveChanges()` will not have been called. If there are no changes, then `result.fields` will be empty, so I'll add in a check for that.